### PR TITLE
perf: update mirrors in a workpool

### PR DESCRIPTION
**Pull request by aymanbagabas**
Implement a simple chunked workqueue to queue updating mirrors. We use the number of cpus to calculate the number of workers to distribute the work to.

### CODE REVIEW

File: server/jobs.go [hunk 1]
```diff
@@ -25,26 +27,36 @@ func (s *Server) mirrorJob() func() {
 			return
 		}
 
+		// Divide the work up among the number of CPUs.
+		wq := sync.NewWorkQueue(runtime.NumCPU() / 4)
```
[the remark this answers is not in the document]
Comment (aymanbagabas) [reply server/jobs.go:31]: didn't want to add to many jobs but i think using the number of cpus as workers is fine and more performant

File: internal/sync/workqueue.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
+package sync
```
[the remark this answers is not in the document]
Comment (aymanbagabas) [reply internal/sync/workqueue.go:1]: should we use errgroup instead?

[the remark this answers is not in the document]
Comment (caarlos0) [reply internal/sync/workqueue.go:1]: I think we can, and then we probably don't need the map either... 

I'll actually move that package to my /sync pkg

[the remark this answers is not in the document]
Comment (caarlos0) [reply internal/sync/workqueue.go:1]: ah, actually, will not, the main part was added into errgroup, you can `SetLimit`

[the remark this answers is not in the document]
Comment (aymanbagabas) [reply internal/sync/workqueue.go:1]: looking at errgroup, it works differently, it blocks if limit is reached so i guess no errgroup

[the remark this answers is not in the document]
Comment (caarlos0) [reply internal/sync/workqueue.go:1]: it blocks when you can `Wait() error` only, `Go(fn() error)` will just add the job to the queue...

**Pull request merged**